### PR TITLE
Add Python 3.13 to CI

### PR DESCRIPTION
**Pull request by Gallaecio**
Current blockers:
- [x] [zstandard](https://github.com/indygreg/python-zstandard/issues/210) (directly and via mitmproxy)
- [x] [pywin32](https://github.com/mhammond/pywin32/issues/2367)
- [x] [Pillow](https://github.com/python-pillow/Pillow/issues/8076)
- [x] [typing-extensions](https://github.com/python/typing_extensions/issues/405)
- [x] [uvloop](https://github.com/MagicStack/uvloop/issues/622)
- [x] [greenlet](https://github.com/python-greenlet/greenlet/pull/396) (via bpython)
- [x] [pyftpdlib](https://github.com/giampaolo/pyftpdlib/issues/650)

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -19,13 +19,15 @@ deps =
     sybil >= 1.3.0  # https://github.com/cjw296/sybil/issues/20#issuecomment-605433422
     testfixtures
     pywin32; sys_platform == "win32"
+    cffi >= 1.17.0rc1; python_version >= '3.13'
```
Comment (wRAR) [inline tox.ini:22]: To remove after a full release is made (planned in July).

File: tox.ini [hunk 2]
```diff
@@ -19,13 +19,15 @@ deps =
...
 
 [testenv]
 deps =
     {[test-requirements]deps}
 
     # mitmproxy does not support PyPy
-    mitmproxy; implementation_name != 'pypy'
+    # mitmproxy requires zstandard which is not yet available on 3.13
+    mitmproxy; implementation_name != 'pypy' and python_version < '3.13'
```
Comment (wRAR) [inline tox.ini:30]: To remove after a full cffi release is made (planned in July) and zstandard wheels are built with it.

File: tox.ini [hunk 3]
```diff
@@ -149,13 +151,13 @@ deps =
...
     robotexclusionrulesparser
-    Pillow
+    Pillow; python_version < '3.13'
     Twisted[http2]
     uvloop; platform_system != "Windows"
     bpython  # optional for shell wrapper tests
     brotli; implementation_name != 'pypy'  # optional for HTTP compress downloader middleware tests
     brotlicffi; implementation_name == 'pypy'  # optional for HTTP compress downloader middleware tests
-    zstandard; implementation_name != 'pypy'  # optional for HTTP compress downloader middleware tests
+    zstandard; implementation_name != 'pypy' and python_version < '3.13'  # optional for HTTP compress downloader middleware tests
```
Comment (wRAR) [inline tox.ini:160]: To remove after a full cffi release is made (planned in July) and zstandard wheels are built with it.

File: tox.ini [hunk 4]
```diff
@@ -149,13 +151,13 @@ deps =
     # restrictions in their deps, so we need to pin old markupsafe here too.
     markupsafe < 2.1.0
     robotexclusionrulesparser
-    Pillow
+    Pillow; python_version < '3.13'
```
Comment (wRAR) [inline tox.ini:154]: To remove after 10.4.0 is released (planned in July).

**Comment by wRAR** [comment]
Looks like `get_func_args()` for `operator.itemgetter(2)` got some new result in 3.13.

**Comment by clin1234** [comment]
pywin32 now supports 3.13

**Comment by wRAR** [comment]
@clin1234 as far as I can see it's still not released though.

**Comment by wRAR** [comment]
> Looks like `get_func_args()` for `operator.itemgetter(2)` got some new result in 3.13.

https://github.com/python/cpython/issues/118285

It now matches the PyPy expectations.

**Comment by wRAR** [comment]
A pywin32 release has unblocked Windows tests and they fail with a permission error in the FTP ones. Which is interesting but I haven't looked closely yet.

**Comment by wRAR** [comment]
I wonder if we should release 2.11.3 with this when it's finished.

**Comment by wRAR** [comment]
Oops, I've forgot that the FTP Windows problem is still unresolved.

**Comment by wRAR** [comment]
pyftpdlib on Windows/Python 3.13 has no trivial fix so I propose merging and even releasing without it.

**Comment by wRAR** [comment]
(I think we want to gracefully handle its absence like with uvloop instead of adding `skipif` like we did for early 3.12, but I'm not sure if a fixture, like the one we have for uvloop, is better than just `importorskip`)

**Comment by Gallaecio** [comment]
+1 to releasing without it. Feel free to follow whichever approach you prefer for test skipping.

**Review by wRAR — APPROVED** [review]

**Pull request merged**